### PR TITLE
Lwm2m shell documentation

### PR DESCRIPTION
**Pull request by VeijoPesonen**
* Adds description off the LwM2M shell functionality.
* Minor fixes to bring the subsystem docs and API docs into line with the specification v1.1

### CODE REVIEW

**Comment by VeijoPesonen** [comment]
@SeppoTakalo , @jheiskan81 , @jarlamsa, @juhaylinen Please review.

**Review by SeppoTakalo — DISMISSED** [review]

**Comment by juhaylinen** [comment]
@VeijoPesonen looks good to me

**Review by jarlamsa — DISMISSED** [review]
LGTM

**Review by rlubos — COMMENTED** [review]
Just some nits, thanks!

File: doc/connectivity/networking/api/lwm2m.rst [hunk 1]
```diff
@@ -517,6 +517,57 @@ The events are prefixed with ``LWM2M_RD_CLIENT_EVENT_``.
...
+
+   uart:~$ lwm2m
+   lwm2m - LwM2M commands
+   Subcommands:
+   exec    :Execute a resource
+            exec PATH
+
+   read    :Read value from LwM2M resource
+            read PATH [OPTIONS]
+            -s   Read value as string(default)
```
[the remark this answers is not in the document]
Comment (VeijoPesonen) [reply doc/connectivity/networking/api/lwm2m.rst]: Fixed.

File: doc/connectivity/networking/api/lwm2m.rst [hunk 2]
```diff
@@ -517,6 +517,57 @@ The events are prefixed with ``LWM2M_RD_CLIENT_EVENT_``.
...
+            -b   Read value as bool (1/0)
+            -uX  Read value as uintX_t
+            -sX  Read value as intX_t
+            -f   Read value as float
+
+   write   :Write into LwM2M resource
+            write PATH [OPTIONS] VALUE
+            -s   Value as string(default)
```
[the remark this answers is not in the document]
Comment (VeijoPesonen) [reply doc/connectivity/networking/api/lwm2m.rst]: Fixed.

File: doc/connectivity/networking/api/lwm2m.rst [hunk 3]
```diff
@@ -517,6 +517,57 @@ The events are prefixed with ``LWM2M_RD_CLIENT_EVENT_``.
...
+            -b   Value as bool
+            -uX  Value as uintX_t
+            -sX  Value as intX_t
+            -f   Value as float
+
+   start   :Start the LwM2M RD (Registration / Discovery) Client
+            start EP_NAME [OPTIONS] [BOOTSTRAP FLAG]
```
[the remark this answers is not in the document]
Comment (VeijoPesonen) [reply doc/connectivity/networking/api/lwm2m.rst]: Brought the help up to date with the current implementation.

**Review by rlubos — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@jukkar Please review.

**Review by jarlamsa — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**